### DISCUSSION TITLE
Wearable system filters

### DISCUSSION THREAD

**Discussion by Asterion1989** [sunflower-land/sunflower-land#2886] [Ideas]
Hello, I had a suggestion about wearable system...

Today is hard to find an specific wearable and to know what kind of buffs are active, so I suggest 3 small changes:

1. Search bar
2. Filter wearables with buff
3. Display active buff

See the [figma prototype here](https://www.figma.com/proto/JzJIqYL8m7w7UYuBHdz9MB/SFL-Farm-Editor-(v2.2)-(Copy)?type=design&node-id=8005-2161&t=w3GEAfksnpQEhDgs-1&scaling=scale-down&page-id=0%3A1&starting-point-node-id=8004%3A1887&mode=design)

![filter](https://github.com/sunflower-land/sunflower-land/assets/78221179/e604192a-c625-4d25-bb2d-68a1f1fceebc)

**Comment by future2back** [comment]
I loved this idea!

If you could create an interface that shows the buffs we are using as illustrated/mentioned on #2849 it would be very interesting.

Thank you.

**Comment by spencerdezartsmith** [comment]
This is great. I have it on my task list to rework this. I would suggest designing a prototype in mobile view also. It's always a challenge to find something that works well on a mobile screen. I would also say that we don't love input boxes mostly due to the keyboard on mobile devices popping up so we use them sparingly.  I am happy to work with you to implement the design once we find one that works :)

If nothing was searched for would you see all items? I think grouping by type is also helpful as you don't always know the name of the item or remember what you have 😅

**Reply by future2back** [reply]
The idea I brought up in #2849 is to avoid clicking on Bumpkin to see the Active buffs.

**Comment by Asterion1989** [comment]
I intend to iterate more soon, and try to bring some ideas that @future2back suggest.

@spencerdezartsmith did u see the figma prototype that I had attached on te post? 
For the mvp of this, I think just the filter "show buff itens" will bring a great value for our players.. and yes, it the button "show buff itens" are off, it will show every wearable possible, the same it was today.

**Reply by Asterion1989** [reply]
@spencerdezartsmith 
![sunflowerfilters](https://github.com/sunflower-land/sunflower-land/assets/78221179/70004973-32a4-48ef-90b3-00351daa7140)

**Comment by future2back** [comment]
Hello @spencerdezartsmith friend. Any news about it?
The current system is messy. Lets try to solve it, please.

Nice job @Asterion1989 !

Thank you.

**Comment by MrSnorch** [comment]
**I like this idea very much, please implement this!**
![image](https://github.com/sunflower-land/sunflower-land/assets/104712725/67076cd9-44a9-434a-8de6-8795b37b3694)

**Reply by future2back** [reply]
Yes, these ideas are essential.

It would be very interesting to have icons that represent each part of the body that the item occupies. I wrote next to the image just for illustration, but it would look really cool if they were icons.

![image](https://github.com/sunflower-land/sunflower-land/assets/74167741/7d0001f7-c7c1-4568-b012-34d1c6ec20ec)

**Filter by tag**


![image](https://github.com/sunflower-land/sunflower-land/assets/74167741/b5fdcf56-77db-4b5f-b9a8-18e6a570c5e9)

In this tag filter, I think it would be interesting to have something that indicates whether you have any item equipped or not in that tag. Some items are mandatory, others are not... I think the proposed filter helps a lot, but it still doesn't fully solve the issue; it still seems a bit confusing to me. I'm an experienced player, but I think new players might find this confusing... Where would a onesie or a coat fit into this filter? Under shirts?


```
HAT
PANTS
AMULET
HAIR
ONESIE
APRON
SHOES
POTION
BEARD
SHIRT
BACKGROUND
1ST HAND
2ND HAND
BACK
```

What is missing from this list? In which section can items still be equipped?

I think we should define and standardize where items can be equipped. Perhaps new players don't know that they can equip, for example, a Shirt, Apron, and Onesie, which visually stack on top of each other.



Examples from mmo inventory:

![image](https://github.com/sunflower-land/sunflower-land/assets/74167741/4464bbc6-6b6d-4b9c-9f39-5203fc0165e0)

![image](https://github.com/sunflower-land/sunflower-land/assets/74167741/9133dec9-69fc-482e-9bdb-ec43c48754aa)

![image](https://github.com/sunflower-land/sunflower-land/assets/74167741/c1c6a30e-40d3-46b2-9861-f68909e50375)

![image](https://github.com/sunflower-land/sunflower-land/assets/74167741/fecd90f3-af13-4068-a9f4-bbe514139a16)

![image](https://github.com/sunflower-land/sunflower-land/assets/74167741/04f2a03d-bcea-49e2-9b24-f8f6234c551a)



By defining where items can be equipped, we can design something similar to the examples above. This would be great for players to know where they have an item and where they don't.

**Reply by Asterion1989** [reply]
Yeah, I love this interface solution. What I had proposed is a MVP, like a Hot fix to be simple to be implemented.

Another kind of solution, is to implement another popup.
for example, if you need to change an iten, you need to select what kind of wearable you want to change, and select the compatible itens in another window.

![image](https://github.com/sunflower-land/sunflower-land/assets/78221179/16824ece-43af-49cc-913c-876936231f92)

I tried to put a section name "appearance" but it consumes many space in the screen on mobile devices... or just I had to add a scroll in this interface

![image](https://github.com/sunflower-land/sunflower-land/assets/78221179/6fd0dd2e-d919-4e8c-919f-d3416c8e6b61)

Or maybe, another tabs

![image](https://github.com/sunflower-land/sunflower-land/assets/78221179/64cd4adb-cdc1-43f8-b1d9-ecd49c08f23d)

**Reply by future2back** [reply]
I loved this solution you presented, my friend, really well done! Congratulations on your work! I think this solves the problem. Besides making the players' lives much easier, it also enhances the game's aesthetics. The current system is too confusing, even for experienced players—imagine for newcomers. I'm sure the solution you proposed is a significant improvement.



![image](https://github.com/sunflower-land/sunflower-land/assets/74167741/4924eb23-f8d4-4376-9611-b0c374437a6d)



I believe that when the player clicks on a particular "slot," it could open a tab, something like what I tried to represent in the design below—apologies for the poor quality. I think you could improve on this and refine the idea further.

![frame93](https://github.com/sunflower-land/sunflower-land/assets/74167741/87d0feea-94f3-4c52-8d44-653fe33bb955)

![frame92](https://github.com/sunflower-land/sunflower-land/assets/74167741/5afb0be0-707e-408a-882e-7f5d6f626cef)



Thank you friend.
